distro/nix/nixos.nix

```diff
@@ -0,0 +1,118 @@
+        systemd = {
+            enable = lib.mkEnableOption "DankMaterialShell systemd startup";
+            restartIfChanged = lib.mkOption {
+                type = bool;
+                default = true;
+                description = "Auto-restart dms.service when dankMaterialShell changes";
+            };
+        };
+        enableSystemMonitoring = lib.mkOption {
+            type = bool;
+            default = true;
+            description = "Add needed dependencies to use system monitoring widgets";
+        };
+        enableClipboard = lib.mkOption {
+            type = bool;
+            default = true;
+            description = "Add needed dependencies to use the clipboard widget";
+        };
+        enableVPN = lib.mkOption {
+            type = bool;
+            default = true;
+            description = "Add needed dependencies to use the VPN widget";
+        };
+        enableBrightnessControl = lib.mkOption {
+            type = bool;
+            default = true;
+            description = "Add needed dependencies to have brightness/backlight support";
+        };
+        enableColorPicker = lib.mkOption {
+            type = bool;
+            default = true;
+            description = "Add needed dependencies to have color picking support";
+        };
+        enableDynamicTheming = lib.mkOption {
+            type = bool;
+            default = true;
+            description = "Add needed dependencies to have dynamic theming support";
+        };
+        enableAudioWavelength = lib.mkOption {
+            type = bool;
+            default = true;
+            description = "Add needed dependencies to have audio wavelength support";
+        };
+        enableCalendarEvents = lib.mkOption {
+            type = bool;
+            default = true;
+            description = "Add calendar events support via khal";
+        };
+        enableSystemSound = lib.mkOption {
+            type = bool;
+            default = true;
+            description = "Add needed dependencies to have system sound support";
+        };
+        quickshell = {
+            package = lib.mkPackageOption pkgs "quickshell" {};
+        };
+    };
+
+    config = lib.mkIf cfg.enable
+    {
+        environment.etc."xdg/quickshell".source = "${dmsPkgs.dankMaterialShell}/quickshell";
+
+        systemd.user.services.dms = lib.mkIf cfg.systemd.enable {
+            Unit = {
+                Description = "DankMaterialShell";
+                PartOf = [config.wayland.systemd.target];
+                After = [config.wayland.systemd.target];
+                X-Restart-Triggers = lib.optional cfg.systemd.restartIfChanged config.programs.quickshell.configs.dms;
+            };
+
+            Service = {
+                ExecStart = lib.getExe dmsPkgs.dmsCli + " run --session";
+                Restart = "on-failure";
+            };
+
+            Install.WantedBy = [config.wayland.systemd.target];
+        };
+
+        environment.systemPackages =
```

Comment:
I think that the basic module options, specially the `enable*` ones should be in a single shared `.nix` file, imported by both home-manager and NixOS modules, since the packages and options are verbatim the same. If someone wanted to add a new dependency or modify the needed, they would always have to unnecessarily have to modify both files.

I can do this in another PR, if you want.